widgets/valitext/org.eclipse.nebula.widgets.valitext/src/org/eclipse/nebula/widgets/TextFieldWithEvaluation/TextFieldWithEvaluation.java

```diff
@@ -0,0 +1,264 @@
+/*******************************************************************************
+* Copyright (C) 2019 Uenal Akkaya <uenal.akkaya@sap.com>, Michael Gutfleisch <michael.gutfleisch@sap.com>
+*
+* This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-2.0/
+* 
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Uenal Akkaya
+*     Michael Gutfleisch
+ ******************************************************************************/
+package org.eclipse.nebula.widgets.TextFieldWithEvaluation;
+
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.events.PaintEvent;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Listener;
+import org.eclipse.swt.widgets.Text;
+
+/**
+ * This widget creates a text field with a build in evaluation of the text.
+ */
+public class TextFieldWithEvaluation extends Composite {
+
+	private Text input;
+	private Label hint;
+	private IEvaluator evaluator;
+
+	/**
+	 * Default constructor. Uses a {@link MaxLengthEvaluator}.
+	 * 
+	 * @param parent
+	 *            The parent composite
+	 * @param maxLength
+	 *            The maximum allowed length of the text
+	 */
+	public TextFieldWithEvaluation(Composite parent, int maxLength) {
```

Comment:
Default widget constructor always must have parent and then the style bit (int). Composite, int with a different meaning is confusing.